src/result.cpp

```
@@ -95,4 +190,455 @@ std::map<Label, std::vector<std::string>> AnalysisResult::find_unreachable(const
+            } else if constexpr (std::is_same_v<T, ZeroCtxOffset>) {
+                return {a.reg};
+            } else if constexpr (std::is_same_v<T, BoundedLoopCount>) {
+                // Loop counter, not a register
+                return {};
+            }
+            return {};
+        },
+        assertion);
+}
+
+std::vector<FailureSlice> AnalysisResult::compute_failure_slices(const Program& prog, const SliceParams params) const {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)